File: Algorithm/Greedy/큰 수 만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    var answer: [String] = []
    var number = number.map{String($0)}, k: Int = k
    
    for n in number {
        while k > 0 && !answer.isEmpty
                && Int(answer.last!)! < Int(n)! {
            answer.removeLast()
            k -= 1
        }
        
        answer.append(n)
    }
    
    while k > 0 {
        answer.removeLast()
        k -= 1
    }
    
    return answer.joined()
}

File: Algorithm/Greedy/큰 수 만들기.py
def solution(number, k):
    answer = ''
    stack = []
    
    for num in number:
        while stack and k > 0 and num > stack[-1]:
            stack.pop()
            k -= 1
        stack.append(num)
    
    if k > 0:
        stack = stack[:len(stack)-k]
    answer = "".join(stack)
    return answer